era/threat/unit.js

```diff
@@ -77,36 +82,47 @@ export class Unit {
       }
     }
     this.buffs = initialBuffs;
-    this.initialCoeff = this.threatCoeff();
+    this.initialCoeff = this.threatCoeff().value;
     if (this.initialCoeff > 1) this.tank = true;
   }
 
+  /**
+   * @param {import("../threat/wcl.js").WCLAbility} [ability]
+   * @returns {import("../base.js").ThreatCoefficient}
+   */
   threatCoeff(ability) {
-    // Ability is of type {type: (int)spellSchool, guid: (int)spellId, [name: string]}
     let spellSchool = ability ? ability.type : this.spellSchool;
     let spellId = ability ? ability.guid : null;
-    let c = this.baseThreatCoeff(spellSchool);
-    for (let i in this.buffs) {
-      if (i in this.config.buffMultipliers) {
-        if (typeof this.config.buffMultipliers[i] === "function") {
-          c *= this.config.buffMultipliers[i](spellSchool);
+    /** @type {import("../base.js").ThreatCoefficient} */
+    let c = applyThreatCoefficient(
+      BASE_COEFFICIENT,
+      this.baseThreatCoeff(spellSchool),
+      `${this.type} (base)`
+    );
+
+    for (let buffId in this.buffs) {
+      if (buffId in this.config.buffMultipliers) {
+        if (typeof this.config.buffMultipliers[buffId] === "function") {
+          const nextC = this.config.buffMultipliers[buffId](spellSchool);
+          c = applyThreatCoefficient(c, nextC, this.config.buffNames[buffId]);
```

Comment:
And here is where we build the unit's current threatCoeff based on buffs, spellId, talents.